sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/models/ComplexField.java

```diff
@@ -0,0 +1,55 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.search.documents.models;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * A helper Field model to build a simple search field.
+ */
+public class ComplexField extends FieldBase {
```

Comment:
Don't use `FieldBase` (at least in .NET, "Base" is not allowed for base classes). See the model rename issue for the suggestion (I forget off hand).

Reply:
name and dataType initialization and validation apply to all three `SimpleField`, `SearchableField`, `ComplexField`
Don't want to repeat them again